patches/server/0807-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,53 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 20 Sep 2021 13:27:50 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index 8a864238e154e2131834d013652746b7e7a78c97..8606a6fa2f1ad2c5fc28c379645131fea7853f7e 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1588,6 +1588,7 @@ public abstract class Mob extends LivingEntity {
+             float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+             if (this.random.nextFloat() < f) {
++                if (!new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), getBukkitEntity()).callEvent()) return; // Paper
+                 player.getCooldowns().addCooldown(Items.SHIELD, 100);
+                 this.level.broadcastEntityEvent(player, (byte) 30);
+             }
+diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
+index cea92f1dc663bf0648b2bd877d86ca380a517bc9..9494f463053d82845c8869d60774f6ab81ffaef6 100644
+--- a/src/main/java/net/minecraft/world/entity/player/Player.java
++++ b/src/main/java/net/minecraft/world/entity/player/Player.java
+@@ -936,7 +936,7 @@ public abstract class Player extends LivingEntity {
+     protected void blockUsingShield(LivingEntity attacker) {
+         super.blockUsingShield(attacker);
+         if (attacker.getMainHandItem().getItem() instanceof AxeItem) {
+-            this.disableShield(true);
++            this.disableShield(true, attacker); // Paper
+         }
+ 
+     }
+@@ -1403,6 +1403,12 @@ public abstract class Player extends LivingEntity {
+     }
+ 
+     public void disableShield(boolean sprinting) {
++        // Paper start
++        disableShield(sprinting, null);
++    }
++
++    public void disableShield(boolean sprinting, @Nullable LivingEntity attacker) {
++        // Paper end
+         float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+         if (sprinting) {
+@@ -1410,6 +1416,8 @@ public abstract class Player extends LivingEntity {
+         }
+ 
+         if (this.random.nextFloat() < f) {
++            org.bukkit.entity.Entity finalAttacker = attacker != null ? attacker.getBukkitEntity() : null; // Paper
```

Comment:
The only time this is null is if something calls the "old" disableShield method. So I would skip calling this event if the attacker paremeter value is null.